include/commissioner/commissioner.hpp

```diff
@@ -127,35 +125,137 @@ struct Config
 };
 
 /**
- * @brief Enumeration of Joiner Type for steering.
+ * @brief The base class defines Handlers of commissioner events.
+ *
+ * Application should inherit this class and override the virtual
+ * functions to provide specific handler.
+ *
+ * @note keep in mind that those handlers will be called in another threads.
```

Comment:
```suggestion
 * @note keep in mind that those handlers will be called in the commissioner thread.
```

Better to explicitly call out the executing thread of handlers?

Reply:
The user may ask what is the **commissioner thread**. I think we'd better hide how we implement the commissioner interface. But, a user must know that the callback is not called in the user's main thread and he/she must add synchronization in the callback if any user data is accessed at there. In this case, whether the **another thread** is **commissioner thread** doesn't matter to the user but will add one more question to him/her. Thoughts?

Reply:
One concern is that users can **NOT** tell from this that they should keep the handlers short (exit quickly) since `another thread` does not necessarily block the commissioner events.

Reply:
Well, I can document **what should not be done in those callbacks**.

Reply:
@simonlingoogle the docstring is updated, please review.